### DISCUSSION TITLE
Allow `odin run - -file`

### DISCUSSION THREAD

**Discussion by mks** [odin-lang/Odin#4587] [Proposals]
Current behaviour:

```sh
❯ cat <<EOF | odin run - -file
package main

import "core:fmt"

main :: proc() {
        fmt.println("Hellope!")
}
EOF
```
Error: `Expected either a directory or a .odin file, got '-'`

It would be nice if it worked.

**Comment by laytan** [comment]
A similar thing that would be nice is allowing `odin run -file path/to/file.odin` which would make shebangs work:
```odin
#!/usr/bin/env odin run -foo -bar -file

package main

import "core:fmt"

main :: proc() {
    fmt.println("Hellope")
}
```

Would allow running it with:
`$ ./path/to/file.odin`

And/or allow leaving out the .odin for the main file which makes it:
`$ ./path/to/file`

**Reply by programandala-net** [reply]
> A similar thing that would be nice is allowing `odin run -file path/to/file.odin` 

Better yet, deprecate `-file`.  I know my file is a file, not a directory; and Odin knows my file is a file, not a directory... so why should I tell Odin my file is a file, not a directory?

Sometime ago I was told that [`-file` has no technical reason, but it's just a "social" option](https://github.com/odin-lang/Odin/discussions/2393).  I understand the rationale but I think it's a non-problem.  Every time I have to write `-file` I feel as if I were writing `-lang:odin` :-)  Also, when I have encouraged other people to try Odin, they were surprised by that unnecessary option and usually forgot to use it during their initial tries.
